Source/WebCore/page/Navigation.cpp

```diff
@@ -569,17 +575,17 @@ void Navigation::resolveFinishedPromise(NavigationAPIMethodTracker* apiMethodTra
         return;
     }
 
-    apiMethodTracker->committedPromise->resolve<IDLInterface<NavigationHistoryEntry>>(*apiMethodTracker->committedToEntry);
-    apiMethodTracker->finishedPromise->resolve<IDLInterface<NavigationHistoryEntry>>(*apiMethodTracker->committedToEntry);
+    Ref { apiMethodTracker->committedPromise }->resolve<IDLInterface<NavigationHistoryEntry>>(Ref { *apiMethodTracker->committedToEntry });
```

Comment:
Please cache `apiMethodTracker->committedToEntry` in a local variable to avoid protecting it twice.